### ISSUE TITLE
getDataset() error

### ISSUE THREAD

**Issue by ArkaprabhaChakraborty** [flow2ml/Flow2ML#106]
### **Describe the bug:**
![Screenshot from 2021-05-21 01-08-09](https://user-images.githubusercontent.com/58482073/119038895-4c0a9f00-b9d1-11eb-9a15-80d1b420c1d1.png)

### **To Reproduce:**
![Screenshot from 2021-05-21 01-05-08](https://user-images.githubusercontent.com/58482073/119038943-5af15180-b9d1-11eb-949e-add04beae24a.png)

### **Expected behavior:**
To create a directory of processed data having training and testing directories.

### **Directory Structure used:**
![Screenshot from 2021-05-21 01-04-36](https://user-images.githubusercontent.com/58482073/119039162-8ffda400-b9d1-11eb-97a4-e8567788bf12.png)

### **Additional context:**

<hr/>

**Note:**
* If you want to work on an issue, you should check if it has already been assigned to anyone.

**Comment by yvkrishna** [comment]
@ArkaprabhaChakraborty, it actually depends on number of images you have in the folders.

How many images fo you have in each label folder ?

**Comment by chebroluharika** [comment]
@ArkaprabhaChakraborty thats because of the value you have taken for train_test_split can’t split properly for the size your training and test dataset size

**Comment by chebroluharika** [comment]
> @ArkaprabhaChakraborty this may occur if you specify wrong path of training dataset.

**Comment by ArkaprabhaChakraborty** [comment]
1000+

**Comment by yvkrishna** [comment]
> 1000+

Images ?

Then check the path.

I don't think we need to give the entire path. Once check it out

**Comment by rubyruins** [comment]
@yvkrishna I think the error could be since no filters or techniques applied. In `create_dataset,` it creates the processedData folder and one by one moves the augmented or filtered images to that folder. In that function, if you print out the value of `filtered_image` it will print out the path of the augmented / filtered images before copying them to processedData folder.

<img width="500" src="https://user-images.githubusercontent.com/50259869/119090279-10032880-ba29-11eb-984b-1e4e8452fc9e.png">

So I think maybe that is why the operation is failing, since no filters or augmentations are applied, the size of processedData is 0 and the dataset is always split from that folder but it has nothing to split. There error says that there are 0 samples to split which occurs when there is no data.

**Comment by chebroluharika** [comment]
@ArkaprabhaChakraborty please try to print self.img_label that you are getting from create_dataset, check if there is any content present there.

**Comment by chebroluharika** [comment]
> @yvkrishna I think the error could be since no filters or techniques applied. In `create_dataset,` it creates the processedData folder and one by one moves the augmented or filtered images to that folder. In that function, if you print out the value of `filtered_image` it will print out the path of the augmented / filtered images before copying them to processedData folder.
> 
> <img alt="" width="500" src="https://user-images.githubusercontent.com/50259869/119090279-10032880-ba29-11eb-984b-1e4e8452fc9e.png">
> 
> So I think maybe that is why the operation is failing, since no filters or augmentations are applied, the size of processedData is 0 and the dataset is always split from that folder but it has nothing to split. There error says that there are 0 samples to split which occurs when there is no data.

yes, according to the pipeline thats there in sample code section, this can be the cause of this bug

**Comment by yvkrishna** [comment]
> @yvkrishna I think the error could be since no filters or techniques applied. In `create_dataset,` it creates the processedData folder and one by one moves the augmented or filtered images to that folder. In that function, if you print out the value of `filtered_image` it will print out the path of the augmented / filtered images before copying them to processedData folder.
> 
> <img alt="" width="500" src="https://user-images.githubusercontent.com/50259869/119090279-10032880-ba29-11eb-984b-1e4e8452fc9e.png">
> 
> So I think maybe that is why the operation is failing, since no filters or augmentations are applied, the size of processedData is 0 and the dataset is always split from that folder but it has nothing to split. There error says that there are 0 samples to split which occurs when there is no data.

Yeah, this is also true.

@ArkaprabhaChakraborty, can you put the screenshot of your code.

Acc to me, 
Cats vs dogs needs data augumentation before training any custom model so try to apply those methods.

Also the example code snippets should contain the main functions we have used.

**Comment by chebroluharika** [comment]
> > @yvkrishna I think the error could be since no filters or techniques applied. In `create_dataset,` it creates the processedData folder and one by one moves the augmented or filtered images to that folder. In that function, if you print out the value of `filtered_image` it will print out the path of the augmented / filtered images before copying them to processedData folder.
> > <img alt="" width="500" src="https://user-images.githubusercontent.com/50259869/119090279-10032880-ba29-11eb-984b-1e4e8452fc9e.png">
> > So I think maybe that is why the operation is failing, since no filters or augmentations are applied, the size of processedData is 0 and the dataset is always split from that folder but it has nothing to split. There error says that there are 0 samples to split which occurs when there is no data.
> 
> Yeah, this is also true.
> 
> @ArkaprabhaChakraborty, can you put the screenshot of your code.
> 
> Acc to me,
> Cats vs dogs needs data augumentation before training any custom model so try to apply those methods.
> 
> Also the example code snippets should contain the main functions we have used.

@yvkrishna Ideally there should not be any tight coupling between code segments. As there may be the scenarios where user may not require augmentations/filters, in which our code doesn't work as expected.  @rubyruins is working on it.

**Comment by yvkrishna** [comment]
> > > @yvkrishna I think the error could be since no filters or techniques applied. In `create_dataset,` it creates the processedData folder and one by one moves the augmented or filtered images to that folder. In that function, if you print out the value of `filtered_image` it will print out the path of the augmented / filtered images before copying them to processedData folder.
> > > <img alt="" width="500" src="https://user-images.githubusercontent.com/50259869/119090279-10032880-ba29-11eb-984b-1e4e8452fc9e.png">
> > > So I think maybe that is why the operation is failing, since no filters or augmentations are applied, the size of processedData is 0 and the dataset is always split from that folder but it has nothing to split. There error says that there are 0 samples to split which occurs when there is no data.
> > 
> > 
> > Yeah, this is also true.
> > @ArkaprabhaChakraborty, can you put the screenshot of your code.
> > Acc to me,
> > Cats vs dogs needs data augumentation before training any custom model so try to apply those methods.
> > Also the example code snippets should contain the main functions we have used.
> 
> @yvkrishna Ideally there should not be any tight coupling between code segments. As there may be the scenarios where user may not require augmentations/filters, in which our code doesn't work as expected. @rubyruins is working on it.

Yeah. The functions should be independent if each other.

@rubyruins go ahead 👍

**Comment by yvkrishna** [comment]
@ArkaprabhaChakraborty, @rubyruins would work on this issue, meanwhile can you try to create results report file for trained pytorch as well ?

**Comment by ArkaprabhaChakraborty** [comment]
Pytorch too 🙂. Okay roger that !!!

**Comment by chebroluharika** [comment]
@yvkrishna please create an issue for that and assign it to @ArkaprabhaChakraborty

**Comment by yvkrishna** [comment]
> @yvkrishna please create an issue for that and assign it to @ArkaprabhaChakraborty

I will do it by tonight.
